traveltimepy/http.py

```diff
@@ -34,12 +38,15 @@ async def send_post_request_async(
     headers: Dict[str, str],
     request: TravelTimeRequest,
     rate_limit: AsyncLimiter,
-) -> T:
+) -> Union[T, TimeMapKmlResponse]:
```

Comment:
This is a very annoying change to me, cause `TimeMapKmlResponse` does meet the requirements of the typevar `T`, but the type checker can't figure that out.
I tried reworking this with abstract `ABC` classes, couldn't fully resolve all type issues. Maybe in the future I'll look into a bigger type rework for this, but that's an issue for a separate PR.

Reply:
I'd like to avoid such hacks

Reply:
yea me too, but I think to avoid it I'd need to make a bigger rework, so wanted to do it in a separate PR

Reply:
what's the problem with TimeMapKmlResponse cause it looks the same as other response classes

Reply:
it's a very weird issue, the type checker can't resolve that TimeMapKmlResponse can fit into T.
This happens because I have a different parsing method for kml (because it's xml and not json). It's hard to explain, I could show you over a call